src/prpy/tsr/tsr.py

```diff
@@ -41,9 +46,23 @@ def __init__(self, T0_w=None, Tw_e=None, Bw=None,
             Tw_e = numpy.eye(4)
         if Bw is None:
             Bw = numpy.zeros((6, 2))
+        if numpy.any(Bw[0:3, 0] > Bw[0:3, 1]):
+            raise ValueError('Bw translation bounds must be [min, max]', Bw)
+
         self.T0_w = T0_w
         self.Tw_e = Tw_e
         self.Bw = Bw
+
+        # We will now create a continuous version of the bound to maintain:
+        # 1. Bw[i,1] > Bw[i,0] which is necessary for LBFGS-B
+        # 2. signed rotations, necessary for expressiveness
+        Bw_cont = numpy.copy(Bw)
+        Bw_cont[3:6, :] = (Bw_cont[3:6, :] + pi) % (2*pi) - pi
```

Comment:
Will this normalization always work? If `Bw_cont[3:6, :] < -pi`, then adding `pi` will not make the value positive. I think this should be:
```python
k = max(int(-Bw_cont[3:6, :].min() / pi), 0) + 1
Bw_cont[3:6, :] = (Bw_cont[3:6, :] + k * pi) % (2*pi) - pi
```
I always screw up angle normalization, so I very well might be wrong here. :sweat: